### DISCUSSION TITLE
How can I forward mentioned messages from a group to different group (bot is admin in both channels)

### DISCUSSION THREAD

**Discussion by CatPeare** [python-telegram-bot/python-telegram-bot#3240] [Q&A] [answered]
**what I want to achieve:** if someone reply to a message using mentions (ex. @admin @mate), the bot will forward this message(s) to a different group.

**my problems are**
bot.forward_message  > this function asking for chat_id, from_chat_id, and message_id > I literally don't understand how to retrieve message_id 

here is my code which makes zero sense. please guide me, because I couldn't find an example of this and I don't know how to use filters as it is supposed to.

```
async def forward_from(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await context.bot.forward_message(chat_id=-1001576505631, from_chat_id=-1001648362294)

if __name__ == '__main__':
    forward_handler = MessageHandler(filters.FORWARDED | filters.TEXT | filters.Entity("@admin"), CallbackContext)

    application.add_handler(forward_handler)
   
    application.run_polling()

```

**Comment by harshil21** [comment, marked as answer]
Filters work by matching the message against it, and if it matches, it sends the update to the handler. You specified `filters.FORWARDED` which means it would accept forwarded messages which is not what you want.

Good thing PTB has shortcut methods so you don't have to type out the message id every time (retrieved by doing `update.message.id`)

``` python
async def forward(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await update.effective_message.forward(chat_id=GROUP_CHAT_ID)

if __name__ == '__main__':
    forward_handler = MessageHandler(filters.REPLY & filters.Text(["@admin", "@mate"]), forward)

    application.add_handler(forward_handler)
   
    application.run_polling()
```
See the [docs](https://docs.python-telegram-bot.org/en/doc-search/telegram.ext.filters.html#telegram.ext.filters.TEXT) if you're confused

**Reply by CatPeare** [reply]
```
in forward
    await update.effective_message.forward(chat_id=-1001648362294, from_chat_id=-100604391079, message_id=update._effective_message.id)
TypeError: Message.forward() got an unexpected keyword argument 'from_chat_id'
```
 
I get this error while trying @admin or @mate
I get the same error if I try `message_id=update.effective_message.id`

**Reply by harshil21** [reply]
you don't have to pass those arguments. Just follow the code snippet

**Reply by CatPeare** [reply]
I couldn't make it run, I tried just like you said and it doesn't give any traceback, also doesn't work. I tried forward_to forward_from all the same. I don't know where I am making a mistake, I even tried using channels names like `chat_id="@mychannel" `

```
async def forward(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await context.bot.forward_message(chat_id=-1001648362294, from_chat_id=update.effective_message.chat_id, message_id=update.effective_message.message_id)
​
if __name__ == '__main__':
    forward_handler = MessageHandler(
        filters.REPLY & filters.Text(["@admin", "@mate"]), forward)
​
    application.add_handler(forward_handler)
​
    application.run_polling()
```

**Reply by harshil21** [reply]
is the bot in the group an admin? Or does the bot have privacy mode turned off? See [this](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-Asked-Questions#what-messages-can-my-bot-see).

Are you replying to any message in the group with the words `@admin/@mate` in them? It's better to also add debug prints in your function to better diagnose it

**Reply by CatPeare** [reply]
it is working like this, I just had to change filters :) thank btw :)

```
async def forward(update: Update, context: ContextTypes.DEFAULT_TYPE):
    await context.bot.forward_message(chat_id="target chat id as int", from_chat_id=update.effective_message.chat_id, message_id=update.effective_message.message_id)
​
​
​
if __name__ == '__main__':
    forward_handler = MessageHandler(
        filters.REPLY | filters.Text(["admin"]) | filters.Regex(r"@admin"), forward)
​
    application.add_handler(forward_handler)
​
    application.run_polling()
```
